dxr/plugins/python/indexers.py

```diff
@@ -153,30 +227,35 @@ def __init__(self, path, contents, plugin_name, tree, python_path,
         self.derived_classes = derived_classes
         self.names = names
 
+        self.module_name = path_to_module(self.python_path, self.path)
+
     def is_interesting(self):
         return is_interesting(self.path)
 
     def needles_by_line(self):
+        self.analyze_tokens()
```

Comment:
What do you think of making analyze_tokens() return its results and changing this statement to assign them to self.node_start_table? I generally like to bubble instance attr assignment up as far as possible to make clear the side effects. Passing the entire file_to_index into the visitor smells a little to me, too, but I can't think of a better idea right now.